controllers/component_build_controller_pac.go

```diff
@@ -148,6 +156,162 @@ func (r *ComponentBuildReconciler) ProvisionPaCForComponent(ctx context.Context,
+		}
+
+		if err := r.Client.Create(ctx, &secret); err != nil {
+			log.Error(err, "failed to create incoming secret", l.Action, l.ActionAdd)
+			return nil, "", false, err
+		}
+
+		reconcileRequired = true
+		log.Info("incoming secret created")
+	}
+	return &secret, incomingSecretName, reconcileRequired, nil
+}
+
+func (r *ComponentBuildReconciler) TriggerPaCBuild(ctx context.Context, component *appstudiov1alpha1.Component) (bool, error) {
+	log := ctrllog.FromContext(ctx).WithName("TriggerPaCBuild")
+	ctx = ctrllog.IntoContext(ctx, log)
+
+	incomingSecret, incomingSecretName, reconcileRequired, err := r.ensureIncomingSecret(ctx, component)
+	if err != nil {
+		return false, err
+	}
+
+	repository, err := r.findPaCRepositoryForComponent(ctx, component)
+	if err != nil {
+		return false, err
+	}
+
+	if repository == nil {
+		return false, fmt.Errorf("PaC repository not found for component %s", component.Name)
+	}
+
+	repoUrl := component.Spec.Source.GitSource.URL
+	gitProvider, err := gitops.GetGitProvider(*component)
+	if err != nil {
+		log.Error(err, "error detecting git provider")
+		// There is no point to continue if git provider is not known.
+		return false, boerrors.NewBuildOpError(boerrors.EUnknownGitProvider, err)
+	}
+
+	pacSecret, err := r.ensurePaCSecret(ctx, component, gitProvider)
+	if err != nil {
+		return false, err
+	}
+
+	gitClient, err := gitproviderfactory.CreateGitClient(gitproviderfactory.GitClientConfig{
+		PacSecretData:             pacSecret.Data,
+		GitProvider:               gitProvider,
+		RepoUrl:                   repoUrl,
+		IsAppInstallationExpected: true,
+	})
+	if err != nil {
+		return false, err
+	}
+
+	// get target branch for incoming hook
+	targetBranch := component.Spec.Source.GitSource.Revision
+	if targetBranch == "" {
+		targetBranch, err = gitClient.GetDefaultBranch(repoUrl)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	if repository != nil {
+		incomingUpdated := updateIncoming(repository, incomingSecretName, pacIncomingSecretKey, targetBranch)
+
+		if incomingUpdated {
+			if err := r.Client.Update(ctx, repository); err != nil {
+				log.Error(err, "failed to update PaC repository with incomings", "PaCRepositoryName", repository.Name)
+				return false, err
+			}
+			log.Info("Added incomings to the PaC repository", "PaCRepositoryName", repository.Name, l.Action, l.ActionUpdate)
+
+			// reconcile to be sure that Repository is updated
+			return true, nil
+		}
+	}
+
+	// reconcile to be sure that Secret is created
+	if reconcileRequired {
+		return true, nil
+	}
+
+	webhookTargetUrl, err := r.getPaCWebhookTargetUrl(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	secretValue := string(incomingSecret.Data[pacIncomingSecretKey][:])
+
+	pipelineRunName := component.Name + pipelineRunOnPushSuffix
+
+	triggerURL := fmt.Sprintf("%s/incoming?secret=%s&repository=%s&branch=%s&pipelinerun=%s", webhookTargetUrl, secretValue, repository.Name, targetBranch, pipelineRunName)
```

Comment:
Probably [url.Values](https://pkg.go.dev/net/url#example-Values) could help here

Reply:
I don't think it is necessary for such single request